modules/tensor_mechanics/src/actions/CohesiveZoneModelAction.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CohesiveZoneModelAction.h"
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+
+registerMooseAction("TensorMechanicsApp", CohesiveZoneModelAction, "add_interface_kernel");
+
+template <>
+InputParameters
+validParams<CohesiveZoneModelAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set cohesive zone model kernels");
```

Comment:
'Action to create an instance of the cohesive zone model kernel for each displacement component'